ibc/client/types/client.go

```diff
@@ -0,0 +1,242 @@
+	consState, err := GetConsensusState(clientStore, height)
+	if err != nil {
+		return 0, err
+	}
+	return consState.GetTimestamp(), nil
+}
+
+// Initialise checks that the initial consensus state is an 08-wasm consensus
+// state and sets the client state, consensus state in the provided client store.
+// It also initializes the wasm contract for the client.
+func (cs *ClientState) Initialise(clientStore modules.ProvableStore, consensusState modules.ConsensusState) error {
+	consState, ok := consensusState.(*ConsensusState)
+	if !ok {
+		return errors.New("invalid consensus state type")
+	}
+	if err := setClientState(clientStore, cs); err != nil {
+		return fmt.Errorf("failed to set client state: %w", err)
+	}
+	if err := setConsensusState(clientStore, consState, cs.GetLatestHeight()); err != nil {
+		return fmt.Errorf("failed to set consensus state: %w", err)
+	}
+	// TODO(#912): implement WASM contract initialisation
+	return nil
+}
+
+//nolint:unused // types defined for future use
+type (
+	verifyMembershipInnerPayload struct {
+		Height           modules.Height            `json:"height"`
+		DelayTimePeriod  uint64                    `json:"delay_time_period"`
+		DelayBlockPeriod uint64                    `json:"delay_block_period"`
+		Proof            []byte                    `json:"proof"`
+		Path             core_types.CommitmentPath `json:"path"`
+		Value            []byte                    `json:"value"`
+	}
+	verifyMembershipPayload struct {
+		VerifyMembership verifyMembershipInnerPayload `json:"verify_membership"`
+	}
+)
+
+// VerifyMembership is a generic proof verification method which verifies a proof
+// of the existence of a value at a given CommitmentPath at the specified height.
+// The caller is expected to construct the full CommitmentPath from a CommitmentPrefix
+// and a standardized path (as defined in ICS 24).
+//
+// If a zero proof height is passed in, it will fail to retrieve the associated consensus state.
+func (cs *ClientState) VerifyMembership(
+	clientStore modules.ProvableStore,
+	height modules.Height,
+	delayTimePeriod, delayBlockPeriod uint64,
+	proof, key, value []byte,
+) error {
+	if cs.GetLatestHeight().LT(height) {
+		return fmt.Errorf("client state height < proof height (%d < %d)", cs.GetLatestHeight(), height)
+	}
+
+	if _, err := GetConsensusState(clientStore, height); err != nil {
+		return errors.New("consensus state not found for proof height")
+	}
+
+	/*
+		payload := verifyMembershipPayload{
+			VerifyMembership: verifyMembershipInnerPayload{
+				Height:           height,
+				DelayTimePeriod:  delayTimePeriod,
+				DelayBlockPeriod: delayBlockPeriod,
+				Proof:            proof,
+				Path:             key,
+				Value:            value,
+			},
+		}
+
+		// TODO(#912): implement WASM contract method calls
+	*/
+
+	return nil
+}
+
+//nolint:unused // types defined for future use
+type (
```

Comment:
Thoughts on moving all the json strutctures into their own file and have the functions be here?